lib/device/fan.js

```diff
@@ -19,32 +33,29 @@ export default class {
     this.enableDebugLogging = accessory.context.enableDebugLogging;
 
     // Add the fan service for the fan if it doesn't already exist
```

Comment:
Here, since you are adding a new service to the accessory, the old service will need to be removed (this check will need to be there every time even though it only needs to happen once really.

```js
// Remove any old original Fan services
if (this.accessory.getService(this.hapServ.Fan)) {
  this.accessory.removeService(this.accessory.getService(this.hapServ.Fan));
}
```

Otherwise for using 'upgrading', they will see two fan icons in the home app, one of them being the `Fan` and the other the `Fanv2`.